### PR TITLE
Improve the service URL builder logic for organization perspective access

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request
 $subject

### Related Issues
- https://github.com/wso2/product-is/issues/17606

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 1]
```diff
@@ -323,7 +324,9 @@ private String buildUrl(String defaultContext, Supplier<String> getValueFromFile
 
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
             try {
-                return ServiceURLBuilder.create().addPath(defaultContext).build().getAbsolutePublicURL();
+                String organizationId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
+                return ServiceURLBuilder.create().addPath(defaultContext).setOrganization(organizationId).build()
+                        .getAbsolutePublicURL();
```
Comment (sadilchamishka) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:329]: Screenshot 2023-11-15 at 07 13 34

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java [hunk 2]
```diff
@@ -470,6 +466,30 @@ protected void appendContextToUri(StringBuilder serverUrl, String contextPath) {
         }
     }
 
+    private boolean setUrlForOrgPerspectiveAccess(StringBuilder resolvedUrlStringBuilder) {
```
Comment (sadilchamishka) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java:469]: add comments, constants

Comment (sadilchamishka) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5148/commits/6919ed3326ae3193a5e6e04f9d09684e8662edf2

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java [hunk 3]
```diff
@@ -466,28 +463,31 @@ protected void appendContextToUri(StringBuilder serverUrl, String contextPath) {
         }
     }
 
-    private boolean setUrlForOrgPerspectiveAccess(StringBuilder resolvedUrlStringBuilder) {
+    private void setUrlBasedOnOrgOrTenantPerspective(StringBuilder resolvedUrlStringBuilder, String tenantDomain) {
```
Comment (sadilchamishka) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java:466]: change method name to setURL

Comment (sadilchamishka) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5148/commits/9210f866727e96ccfb5abef34fc2eab3b90ac60e

**Review by dewniMW — APPROVED** [review]

**Pull request merged**